biorbd_optim/objective_functions.py

```diff
@@ -61,8 +61,8 @@ def minimize_torque(ocp, nlp, weight=1, controls_idx=(), data_to_track=()):
         for i in range(nlp["ns"]):
             ocp.J += (
                 casadi.dot(
-                    nlp["U"][i][controls_idx] - data_to_track[i, controls_idx],
-                    nlp["U"][i][controls_idx] - data_to_track[i, controls_idx],
+                    nlp["U"][i][controls_idx] - data_to_track[i, :],
+                    nlp["U"][i][controls_idx] - data_to_track[i, :],
```

Comment:
Why did you remove the controls_idx?

Reply:
data_to_track has the shape of the number of control you want to minimize. I wanted to minimize the last three torques so if I used the values of control_idx it was out of range

Reply:
That means your data don't fit you control no? U and data_to_track should have the same dimensions. Without control_idx, how can you just minimize the last control for example?

Reply:
In my case, I have 6 torques, to minimize only the last three I put control_idx =[3,4,5]. In the objective function the size of data_to_track is [number of shooting point, len(control_idx)=3] so if I use the values of control_idx it was out of range for data_to_track

Reply:
> data_to_track has the shape of the number of control you want to minimize.
I don't think it is much intuitive... since you are referring to the indices when you declare the objective (`"controls_idx": [some values]`) (if not passed then all the data are used. 

You method could make sense, but it requires to change the input (`"data_to_track`") instead of actually using the indices.

Reply:
Done